.github/workflows/build_and_deploy_documentation.yml

```diff
@@ -0,0 +1,36 @@
+name: "Build Sphinx Documentation"
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main 
+      - ci/setup
+
+permissions:
+  contents: write
```

Comment:
Why is this necessary? Because we are pushing content to the branch "gh-pages"? If I understood this setting correctly, it indicates the permission to create "git-objects"

Reply:
Yes, because we want to create/write files in gh-pages